esp/esp/web/views/myesp.py

```diff
@@ -264,6 +264,9 @@ def myesp_onsite(request):
     # - Currently reverse the list in Python, otherwise fbap's cache is ignored
     # TODO: Fix this
     progs = list(progs)
+    if len(progs) > 0:
+        # filter out past programs; admins can always go to those through /manage/programs to access their onsite pages
+        progs = [p for p in progs if p.datetime_range() is None or len(p.datetime_range()) == 0 or p.datetime_range()[-1] > datetime.datetime.now()]
```

Comment:
Do we really want to include programs without dates?

Reply:
Yeah, I thought about it, and for test programs or programs where people manage to goof something up, I think we still want admins to have full access

Reply:
But if a program doesn't have dates, will any of the onsite stuff even make sense/work?